### PR TITLE
Finish models implementation

### PR DESCRIPTION
**Pull request by advaypal**
The following changes have been made in this PR:

1. A new SessionName model, which serves as a wrapper around the session name string. (Reason explained later)
2. A getPad() function for the Session class
3. Complete implementations for all DataManager methods. Still testing and writing unit tests though.
4. Change the DataManager class to a Singelton

The idea is that when the DataManager instance is created, all the animation strings, audio strings, and session names are loaded and stored in a data structure. Further accesses to the database only take place on adding/deleting new sessions, audios or animations.

### CODE REVIEW

File: MIDIchlorians/MIDIchlorians/DataManager.swift [hunk 1]
```diff
@@ -11,26 +11,109 @@ import RealmSwift
 class DataManager {
 
     private let realm: Realm?
+    static let dataManagerInstance = DataManager()
```
Comment (ngzhian) [inline MIDIchlorians/MIDIchlorians/DataManager.swift]: name this instance, repeating `datamanager` a bit too many times

**Review by ngzhian — APPROVED** [review]
LGTM

**Comment by ngzhian** [comment]
Due to #24 , this has some small conflicts, i guess we can keep both methods around, and will deprecate the one i wrote in a future PR (not a big deal) thanks!

File: MIDIchlorians/MIDIchlorians/SessionName.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+//
+//  Created by Advay Pal on 25/03/17.
+//  Copyright © 2017 nus.cs3217.a0118897. All rights reserved.
+//
+
+import RealmSwift
+
+class SessionName: Object {
+
+    private var sessionName: String = ""
```
Comment (ngzhian) [inline MIDIchlorians/MIDIchlorians/SessionName.swift:13]: if the intention is to make a public getter and a private setter we should use `private(set) var sessionString: String = ""`, that way we won't need the getter below

Comment (advaypal) [reply MIDIchlorians/MIDIchlorians/SessionName.swift:13]: Currently I'm assigning the value via a convenience initaliser, so no setter

**Pull request merged**